### PR TITLE
Refactor BuildDevint with Gradle and Maven wrapper

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        executable mavenExec
+        args mavenArgs(configure([:], mavenClosure))
+    }
+}
+
+task buildUtils {
+    doLast {
+        logger.info('Building Utils ...')
+        maven {
+            goals = "install"
```
Comment (Flanker32) [inline build.gradle]: Shall we execute clean in maven build?

Comment (wezhang) [reply build.gradle]: I'll add an option to control that.

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import org.gradle.internal.os.OperatingSystem
+
+def gradleExec = file(OperatingSystem.current().isWindows() ? "gradlew.bat" : "gradlew")
+def mavenExec = file(OperatingSystem.current().isWindows() ? "mvnw.cmd" : "mvnw")
+def IJ_VERSION_LATEST = IJ_VERSION_MINOR?.trim()?.isEmpty() ? IJ_VERSION_MAJOR : "${IJ_VERSION_MAJOR}.${IJ_VERSION_MINOR}"
+
+def mavenArgs = { conf ->
```
Comment (andxu) [inline build.gradle]: add -B by default

Comment (wezhang) [reply build.gradle]: Fix it!

**Pull request merged**